assets/l10n/app_en.arb

```diff
@@ -308,6 +308,10 @@
   "@loginServerUrlInputLabel": {
     "description": "Input label in login page for Zulip server URL entry."
   },
+  "serverUrlDocLinkLabel": "What's this?",
```

Comment:
```
login: Link to doc for what "server URL" is and how to find it

- Add informative helper text below the "server URL" field in the login screen.
- When tapped, the helper text opens Zulip documentation explaining server URLs and how to find them.
- Improves user experience during login by providing clear guidance.
- Add i18n for "What's this?" helper text.
- Fixes: #109
```

Commit message nit:

Separate the "Fixes" line from the bulleted list. In fact, I think the "Fixes" line adequately explains what's being done and why, so I would delete the rest of the body:

```
login: Link to doc for what "server URL" is and how to find it

Fixes: #109
```